compose/volume.py

```diff
@@ -1,14 +1,23 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
 
+from docker.errors import NotFound
+
 
 class Volume(object):
-    def __init__(self, client, project, name, driver=None, driver_opts=None):
+    def __init__(self, client, project, name, driver=None, driver_opts=None,
+                 external=False):
         self.client = client
         self.project = project
         self.name = name
         self.driver = driver
         self.driver_opts = driver_opts
+        self.external_name = None
+        if external:
+            if isinstance(external, dict):
+                self.external_name = external.get('name')
+            else:
+                self.external_name = self.name
```

Comment:
I wonder if we should do this normalization in the config module, since we do the other related operations there.